src/batch.js

```diff
@@ -36,12 +36,23 @@ function makeBatchRoutingPathFunction(originalUrl, serverURL, publicServerURL) {
   if (serverURL && publicServerURL && serverURL.path != publicServerURL.path) {
     const localPath = serverURL.path;
     const publicPath = publicServerURL.path;
+
     // Override the api prefix
     apiPrefix = localPath;
     return function (requestPath) {
+      // Figure out which server url was used by figuring out which
+      // path more closely matches requestPath
+      const startsWithLocal = requestPath.startsWith(localPath);
+      const startsWithPublic = requestPath.startsWith(publicPath);
+      const pathLengthToUse =
+        startsWithLocal && startsWithPublic
+          ? Math.max(localPath.length, publicPath.length)
```

Comment:
It took me a second to figure out what this was doing. Good way to tell which one "startsWith" more.

Reply:
Sorry about this... I wrote this four times trying to find a clear way to do it and am still unhappy with the readability... definitely open to suggestions on strategy or clarity here!